salt/cert/init.sls

```diff
@@ -7,3 +7,29 @@ include:
          pillar['ssl']['crt_file'],
          pillar['ssl']['key_file'],
          o = pillar['certificate_information']['subject_properties']['O']) }}
+
+#######################################
+# additional system wide certificates #
+#######################################
+
+# Install additional certificates that were setup in Velum by the user as
+# system-wide certificates
+
+{% set system_certs = salt.caasp_pillar.get('system_certificates', []) %}
+{% for cert in system_certs %}
+  {% set name, cert = cert['name'], cert['cert'] %}
+
+/etc/pki/trust/anchors/{{ name }}.crt:
+  file.managed:
+    - makedirs: True
+    - user: root
+    - group: root
+    - mode: 644
+    - contents: |
+        {{ cert | indent(8) }}
+  cmd.run:
+    - name: update-ca-certificates
+    - onchanges:
+        - file: /etc/pki/trust/anchors/{{ name }}.crt
+
+{% endfor %}
```

Comment:
I think that, if you want to run the `update-ca-certificates` only once after several changes, you could achieve it with something like:

```SaltStack
{% set system_certs = salt.caasp_pillar.get('system_certificates', []) %}
{% for cert in system_certs %}
  {% set name, cert = cert['name'], cert['cert'] %}

/etc/pki/trust/anchors/{{ name }}.crt:
  file.managed:
    - makedirs: True
    - user: root
    - group: root
    - mode: 644
    - contents: |
        {{ cert | indent(8) }}
    - onchanges_in:
        - update-ca-certificates

{% endfor %}

update-ca-certificates:
  cmd.run

```

(This is not tested). This should insert a `onchanges` in the `update-ca-certificates` blocks for each certificate we write...

Reply:
@ellisab has tested this scenario in https://github.com/kubic-project/salt/pull/537 and that didnt work. you probably need a new sls with what you suggested in https://github.com/kubic-project/salt/pull/537#discussion_r189196888

Reply:
Might be related to saltstack/salt#19592. Maybe you need to use `watch` instead of `onchanges`.